### PR TITLE
Get allowedPaymentModes from offline service

### PR DESCRIPTION
**Pull request by yksolanki9**

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -1007,24 +1007,13 @@ export class AddEditExpensePage implements OnInit {
   }
 
   getPaymentModes(): Observable<AccountOption[]> {
-    const accounts$ = this.isConnected$.pipe(
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:1010]: Removing this off as it is handled in offline service

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -1007,24 +1007,13 @@ export class AddEditExpensePage implements OnInit {
...
-      take(1),
-      switchMap((isConnected) => {
-        if (isConnected) {
-          return this.accountsService.getEMyAccounts();
-        } else {
-          return this.offlineService.getAccounts();
-        }
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:1017]: Removing this off as it is handled in offline service

File: src/app/core/services/accounts.service.ts [hunk 3]
```diff
@@ -104,16 +104,13 @@ export class AccountsService {
     );
   }
 
-  //Dummy method - will be replaced by API call
-  getAllowedPaymentModes(): Observable<string[]> {
-    return of(['PERSONAL_ADVANCE_ACCOUNT']);
-  }
```
Comment (yksolanki9) [inline src/app/core/services/accounts.service.ts:110]: Moved this method to offline service

File: src/app/core/services/offline.service.ts [hunk 4]
```diff
@@ -416,20 +416,11 @@ export class OfflineService {
...
-  setActiveExpenseTab(activeTab) {
-    return from(this.storageService.set('activeExpenseTab', activeTab));
-  }
-
-  setActiveCorporateCardExpenseTab(activeTab) {
-    return from(this.storageService.set('activeCorporateCardExpenseTab', activeTab));
-  }
-
-  getActiveCorporateCardExpenseTab() {
-    return from(this.storageService.get('activeCorporateCardExpenseTab'));
```
Comment (yksolanki9) [inline src/app/core/services/offline.service.ts:432]: Removed all unused methods

File: src/app/core/services/accounts.service.ts [hunk 5]
```diff
@@ -123,21 +120,19 @@ export class AccountsService {
...
+      etxn,
+      isMileageOrPerDiemExpense
+    );
 
-        return allowedAccounts.map((account) => ({
-          label: account.acc.displayName,
-          value: account,
-        }));
-      })
+    return of(
```
Comment (Julias0) [inline src/app/core/services/accounts.service.ts:131]: Why return this as an observable when we can return it synchronously? Change the return type and remove off the of

**Review by Julias0 — CHANGES_REQUESTED** [review]
1 important comment

File: src/app/core/services/accounts.service.ts (hunk 5 above)
Comment (yksolanki9) [reply src/app/core/services/accounts.service.ts]: Done

**Review by Julias0 — APPROVED** [review]

**Review by gokulk16 — APPROVED** [review]

**Pull request merged**